--- a/src/linux_joystick.c
+++ b/src/linux_joystick.c
@@ -25,6 +25,9 @@
 //
 //========================================================================

+#define _XOPEN_SOURCE 700
+#define _GNU_SOURCE
+
 #include "internal.h"

 #if defined(GLFW_BUILD_LINUX_JOYSTICK)
--- a/src/posix_time.c
+++ b/src/posix_time.c
@@ -25,6 +25,9 @@
 //
 //========================================================================

+#define _POSIX_SOURCE 199309L
+#define _POSIX_C_SOURCE 199309L
+
 #include "internal.h"

 #if defined(GLFW_BUILD_POSIX_TIMER)

--- a/src/win32_module.c
+++ b/src/win32_module.c
@@ -34,7 +34,7 @@

 void* _glfwPlatformLoadModule(const char* path)
 {
-    return LoadLibraryA(path);
+    return LoadLibraryExA(path, NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
 }

 void _glfwPlatformFreeModule(void* module)
