diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea5a6bfe..e50987d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,6 +48,14 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/openvkl/include/${PROJECT_NAME}/versio
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}
 )
 
+set(CMAKE_SKIP_INSTALL_RPATH OFF)
+if (APPLE)
+  set(CMAKE_MACOSX_RPATH ON)
+  set(CMAKE_INSTALL_RPATH "@loader_path/")
+else()
+  set(CMAKE_INSTALL_RPATH "\$ORIGIN")
+endif()
+
 ## openvkl specific configuration ##
 
 option(BUILD_EXAMPLES "Build example applications." ON)
