diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4acb869..b27abfc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,6 +46,14 @@ option(BUILD_SHARED_LIBS "Build rkcommon as a shared library" ON)
 option(RKCOMMON_ADDRSAN "Build rkcommon with dlclose disabled for addrsan" OFF)
 option(RKCOMMON_NO_SIMD "Build rkcommon not using SIMD instructions" OFF)
 
+set(CMAKE_SKIP_INSTALL_RPATH OFF)
+if (APPLE)
+  set(CMAKE_MACOSX_RPATH ON)
+  set(CMAKE_INSTALL_RPATH "@loader_path/")
+else()
+  set(CMAKE_INSTALL_RPATH "\$ORIGIN")
+endif()
+
 include(CTest)
 
 if (BUILD_TESTING)
