commit 27f34a580be4a3becf5f8c0cba13433f53c21337
Author: Richard O'Grady <rjogrady@google.com>
Date:   Wed Jul 12 19:12:01 2023

    Fix -Wsign-compare warning
    
    PiperOrigin-RevId: 547529709

diff --git a/snappy.cc b/snappy.cc
index 688065b..6473123 100644
--- a/snappy.cc
+++ b/snappy.cc
@@ -1289,7 +1289,7 @@ std::pair<const uint8_t*, ptrdiff_t> DecompressBranchless(
         DeferMemCopy(&deferred_src, &deferred_length, from, len);
       }
     } while (ip < ip_limit_min_slop &&
-             (op + deferred_length) < op_limit_min_slop);
+             static_cast<ptrdiff_t>(op + deferred_length) < op_limit_min_slop);
   exit:
     ip--;
     assert(ip <= ip_limit);
